linkerd/app/src/env.rs

```diff
@@ -171,6 +174,7 @@ const DEFAULT_INBOUND_CONNECT_BACKOFF: ExponentialBackoff = ExponentialBackoff {
 };
 const DEFAULT_OUTBOUND_DISPATCH_TIMEOUT: Duration = Duration::from_secs(3);
 const DEFAULT_OUTBOUND_CONNECT_TIMEOUT: Duration = Duration::from_secs(1);
+const DEFAULT_PROTOCOL_DETECT_TIMEOUT: Duration = Duration::from_secs(10);
```

Comment:
What do you think about using the DISPATCH_TIMEOUT settings to control the detect timeout. Specifically, I'm fine with them being distinct variables in the config objects, but from the operator's point of view, are these things materially different? They are both effectively "the amount of time I'm willing to spend in the proxy before giving up".

Reply:
:+1:, I'm not convinced these need to be separately tuneable knobs?